tools/block-generator/generator/generate.go

```diff
@@ -410,20 +418,25 @@ func (g *generator) WriteBlock(output io.Writer, round uint64) error {
 	if err != nil {
 		return err
 	}
-
+	g.ledger.AddBlock(cert.Block, agreement.Certificate{})
```

Comment:
I ended up using AddBlock() instead of using eval.Eval() since AddBlock calls eval.Eval already. And state delta can be retrieved using `ledger.GetStateDeltaForRound(basics.Round(round))`